File: Java/src/main/java/QuestionBank/Solution801_900/Solution830.java
package QuestionBank.Solution801_900;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

/**
 * @author windows
 */
public class Solution830 {
    public List<List<Integer>> largeGroupPositions(String s) {
        int count = 1;
        ArrayList<List<Integer>> res = new ArrayList<>();
        for (int i = 0; i < s.length() - 1; i++) {
            if (s.charAt(i) == s.charAt(i + 1)){
                count++;
            }else{
                if(count >= 3){
                    res.add(Stream.of(i - count + 1, i).collect(Collectors.toList()));
                }
                count = 1;
            }
        }
        if (count >= 3){
            res.add(Stream.of(s.length() - count, s.length() - 1).collect(Collectors.toList()));
        }
        return res;
    }


    public static void main(String[] args) {
        Solution830 solution830 = new Solution830();
    }
}

File: C++/src/Solution801_900/Solution830.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution830.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<vector<int>> largeGroupPositions(string s) {
  s = s + "A";
  int n = s.size();
  vector<vector<int>> res;
  int count = 1;
  for (int i = 0; i < n; ++i) {
    if (i + 1 < n){
      if (s[i] == s[i + 1]){
        count++;
      } else{
        if (count >= 3){
          res.push_back({i - count + 1, i});
        }
        count = 1;
      }
    }
  }
  return res;
}